### ISSUE TITLE
Same US capital gains taxation problems when upgrading from 0.118.0 to 0.121.1

### ISSUE THREAD

**Issue by martinholmer** [PolicyEngine/openfisca-us#1219]
The 2287 US income tax differences in the **t21** sample under version 0.118.0 continue to appear after upgrading to the 0.121.1 version of OpenFisca-US.  However, after the upgrade the size of these US tax differences do change, but the largest differences *increase* rather than decrease in size.

Consider the case reported in issue #1195, which under version 0.118.0 had a tax difference of $1200.  Now, under version 0.121.1, this case has a difference of $1500.  Here is the test case and the new test results for the case:
```
% cat t21-id95009.test.yaml
(tafe) US% ../show_output.sh t21 95009
>>> OpenFisca-US YAML test input:
# OpenFisca-US test file derived from t21.its.csv and t21.ota.csv files

- name: Tax unit with taxsimid 95009 from t21.its.csv
  absolute_error_margin: 0.01
  period: 2021
  input:
    people:
      person1:
        is_tax_unit_head: 1
        is_tax_unit_spouse: 0
        is_tax_unit_dependent: 0
        age: 40
        employment_income: 49000
        taxable_interest_income: 9000
        long_term_capital_gains: 10000
        social_security: 21000
        real_estate_taxes: 17000
        interest_expense: 25000
        ssi: 0
        state_supplement: 0
        wic: 0
      person2:
        is_tax_unit_head: 0
        is_tax_unit_spouse: 0
        is_tax_unit_dependent: 1
        age: 11
        ssi: 0
        state_supplement: 0
        wic: 0
    spm_units:
      spm_unit:
        members: [person1, person2]
        snap: 0
        tanf: 0
    tax_units:
      tax_unit:
        members: [person1, person2]
        tax_unit_childcare_expenses: 9000
        premium_tax_credit: 0
    households:
      household:
        members: [person1, person2]
        state_code: TX
  output:
    taxsim_tfica: 3748.50
    income_tax: -5182.00
```
The test results for the above case are:
```
================================= FAILURES ===================================
_______________________________ test session _________________________________
/Users/mrh/work/OpenFisca-US/US/t21-id95009.test.yaml:
  Test 'Tax unit with taxsimid 95009 from t21.its.csv':
    income_tax@2021: [-3682.] differs from -5182.0
      with an absolute margin [1500.] > 0.01
========================= short test summary info ============================
FAILED t21-id95009.test.yaml::
============================ 1 failed in 1.10s ===============================
```

Given that @nikhilwoodruff said in issue #1195 that he could eliminate the US income tax difference (only?) if he changed the $25,000 mortgage interest expense to zero suggests the tax difference may be related to how the tax laws are being applied in the presence of long-term capitals gains and itemized deduction expenses.  For what it is worth, the trace output shows the `capital_gains_tax` being $1500, which appears to be the correct amount for the $10,000 in long-term capital gains.

The $1500 difference shows up in `income_tax_before_credits` variable, which in the trace output is $6118 versus the $4618 amount for the corresponding variable in TAXSIM35.  The two models agree on `taxable_income`.  Is it possible that the $1500 tax on long-term capital gains is being (inadvertently) added twice?  Or is the tax difference being equal to the size of the tax on long-term capital gains just a coincidence?

**Comment by martinholmer** [comment]
Issue #1219 reported 2287 US income tax differences in the **t21** sample.  When I specify a slightly different sample, which is the same as the **t21** sample except there are no child dependents (and therefore, no tax units have the HEAD_OF_HOUSEHOLD filing status), all the 2287 tax differences go away.

This finding suggests that the parameter bug reported in the issue #1222 [conversation](https://github.com/PolicyEngine/openfisca-us/issues/1222#issuecomment-1207406996), is the cause of all of the US income tax differences reported in issue #1219.

**Comment by martinholmer** [comment]
Issue #1219 has been resolved in the 0.121.2 version.
